packages/experience-builder-sdk/src/hooks/useFetchById.spec.ts

```diff
@@ -0,0 +1,168 @@
+import { useFetchById, UseFetchByIdArgs } from './useFetchById';
+import { EntityStore } from '@contentful/experience-builder-core';
+import { renderHook, waitFor } from '@testing-library/react';
+import { compositionEntry } from '../../test/__fixtures__/composition';
+import { entries, assets } from '../../test/__fixtures__/entities';
+import type { ContentfulClientApi, Entry } from 'contentful';
+import type { ExternalSDKMode } from '@contentful/experience-builder-core/types';
+
+const experienceTypeId = 'layout';
+const localeCode = 'en-US';
+const id = 'composition-id';
+
+let clientMock: ContentfulClientApi<undefined>;
+
+describe('useFetchById', () => {
+  beforeEach(() => {
+    clientMock = {
+      getEntries: jest.fn().mockImplementation((data) => {
+        if ('sys.id[in]' in data) {
+          return Promise.resolve({ items: entries });
+        }
+
+        return Promise.resolve({ items: [compositionEntry] });
+      }),
+      getAssets: jest.fn().mockResolvedValue({ items: assets }),
+    } as unknown as ContentfulClientApi<undefined>;
+  });
+
+  it('should be defined', async () => {
+    const { result } = renderHook(useFetchById, {
+      initialProps: {
+        client: clientMock,
+        mode: 'preview' as ExternalSDKMode,
+        id,
+        experienceTypeId,
+        localeCode,
+      },
+    });
+
+    await waitFor(() => {
+      expect(result.current).toBeDefined();
+      expect(result.current.isLoading).toBe(true);
+    });
```

Comment:
[nitpick]
`waitFor` is not meant to be used with multiple assertions and can lead to unwanted issues:
```suggestion
    await waitFor(() => expect(result.current?.isLoading).toBe(true));
```
See here for more explanation on this: https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#having-multiple-assertions-in-a-single-waitfor-callback